core/platform/auth/gae_auth_services.py

```diff
@@ -0,0 +1,118 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service layer for handling user-authentication with GAE's built-in system."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import auth_domain
+from core.platform import models
+import python_utils
+
+from google.appengine.api import users
+
+auth_models, = models.Registry.import_models([models.NAMES.auth])
+
+
+def create_user_auth_details(user_id, auth_id):
+    """Returns a UserAuthDetails object configured with GAE properties."""
+    return auth_domain.UserAuthDetails(user_id, gae_id=auth_id)
+
+
+def get_auth_claims_from_request(unused_request):
+    """Returns Claims for the user currently signed-in and sending requests.
+
+    Args:
+        unused_request: webapp2.Request. Unused because Google AppEngine handles
+            user authentication internally.
+
+    Returns:
+        AuthClaims|None. Claims about the currently signed in user. If no user
+        is signed in, then returns None.
+    """
+    gae_user = users.get_current_user()
+    if gae_user is not None:
+        return auth_domain.AuthClaims(
+            gae_user.user_id(), gae_user.email(), users.is_current_user_admin())
+    return None
+
+
+def disable_auth_associations(user_id):
+    """Disable the auth associations of the given user so they can't be used."""
+    assoc_model = auth_models.UserIdentifiersModel.get_by_user_id(user_id)
+    if assoc_model is not None:
+        assoc_model.deleted = True
+        assoc_model.update_timestamps()
+        assoc_model.put()
+
+
+def delete_auth_associations(unused_user_id):
+    """No special action is necessary for deleting Google AppEngine users."""
+    pass
+
+
+def are_auth_associations_deleted(unused_user_id):
+    """No special action is necessary for deleting Google AppEngine users."""
+    return True
+
+
+def get_user_id_from_auth_id(auth_id):
+    """Returns the user ID associated with the given auth ID."""
```

Comment:
Here and elsewhere, please make sure all your docstrings have args/returns/raises where appropriate.

/cc @DubeySandeep -- this is something we may want to prioritize for lint checks.